### ISSUE TITLE
Color map

### ISSUE THREAD

**Issue by Kandob** [wave-harmonic/crest#379]
Hello, is it possible to add a color map to have parts of the ocean with different colors? For example, having blue water that transitions to green near a swamp. Tiling a texture, vertex painting or a world aligned plane that projects a texture would be great.

**Comment by huwb** [comment]
Yeah it would be a case of adding another texture like the current foam  / caustics texture and sampling it to get the colour.

I'll get onto this at some point, but ill mark it as help wanted as it could be a good exercise for someone else to try if they want.

**Comment by Kandob** [comment]
I've been able to get a decent result by using the render alpha on surface feature. The lighting is not entirely correct but it will do for now. I'm using a color texture with an alpha now but the ability to multiply a color in the shader would be a great addition, that way I can make the texture grayscale and pick a color in the shader.

![Annotation 2019-12-23 154427](https://user-images.githubusercontent.com/14958256/71364012-3d7d4580-259b-11ea-8962-8d1a119bd670.png)

**Comment by ProtopottyGames** [comment]
I feel compelled to mention what I would probably do in this particular case. Not entirely sure whether it would work for the OP's situation, but I'm a big fan of changing things based on distance. Kinda like how a biome system works with weather. The closer you get to a swamp, the more it starts to look like a swamp. The closer you get to a coast, the less it looks like whatever your character is walking away from and the more it starts to look like a coast etc.. 
If your gameplay permits this kind of "cross-fading" it could end up being a lot less work and you may even get some cool looking transitions. Depending on how you set it up it may even save on performance.

**Comment by magenta404** [comment]
Closing as implemented in Crest 5.
